exasol_script_languages_developer_sandbox/lib/asset_printing/print_assets.py

```diff
@@ -143,6 +143,8 @@ def print_s3_objects(aws_access: AwsAccess, asset_id: Optional[AssetId], printin
 
     s3_objects = aws_access.list_s3_objects(bucket=vm_bucket, prefix=BUCKET_PREFIX)
     if len(prefix) > 0:
+        if prefix[-1] != "*":
+            prefix = f"{prefix}*"
```

Comment:
this I don't understand, does make a * to a recursive wildcard **

Reply:
I think this code deserves some documentation, indeed.

I added the following documentation to the code:

```

    # How the filtering works:
    # 1. The VM are stored under following location in the S3 Bucket: $BUCKET_PREFIX/$AssetId/name.$VM_FORMAT
    #    For example "slc_developer_sandbox/5.0.0/export-ami-01be860e6a6a98bf8.vhd"
    # 2. Because S3 list_s3_object does not support we need to implement our own wildcard implementation here.
    #    We cal list_s3_object with the standard prefix (e.g. "slc_developer_sandbox"),
    #    which returns ALL stored vm objects.
    # 3. If no filter is given at command line variable "prefix" will be empty and we return all s3 objects
    # 4. If variable "prefix" is not empty, we need to ensure that it end's with a wildcard, so that the matching
    #    works corecty.
    # => Assume that a filter is given  "5.0.0". Variable prefix would be "slc_developer_sandbox/5.0.0". 
    ```